base/test/affinetransform_tests.cpp

```diff
@@ -0,0 +1,390 @@
+
+	f.fileReader->step();
+	f.copy1->step();
+	f.affineTransform->step();
+	f.copy2->step();
+
+	auto outputPinId = f.copy2->getAllOutputPinsByType(FrameMetadata::RAW_IMAGE)[0];
+	auto frames = f.m3->pop();
+	BOOST_TEST((frames.find(outputPinId) != frames.end()));
+	auto outFrame = frames[outputPinId];
+	BOOST_TEST(outFrame->getMetadata()->getFrameType() == FrameMetadata::RAW_IMAGE);
+	Test_Utils::saveOrCompare("./data/testOutput/affinetransform_tests_frame_all_400x400_rgba.raw", (const uint8_t*)outFrame->data(), outFrame->size(), 0);
+}
+	
+BOOST_AUTO_TEST_CASE(BGRA_Image_shift_scale_rotate)
+{
+	ImageMetadata::ImageType::BGRA;
+	AffineTestsStruct f("./data/8bit_frame_1280x720_bgra.raw", 1280, 720, ImageMetadata::ImageType::BGRA, CV_8UC4, AffineTransformProps::LINEAR, 90, 100,100, 2.5f);
+
+	f.fileReader->step();
+	f.copy1->step();
+	f.affineTransform->step();
+	f.copy2->step();
+
+	auto outputPinId = f.copy2->getAllOutputPinsByType(FrameMetadata::RAW_IMAGE)[0];
+	auto frames = f.m3->pop();
+	BOOST_TEST((frames.find(outputPinId) != frames.end()));
+	auto outFrame = frames[outputPinId];
+	BOOST_TEST(outFrame->getMetadata()->getFrameType() == FrameMetadata::RAW_IMAGE);
+	Test_Utils::saveOrCompare("./data/testOutput/affinetransform_tests_8bit_frame_1280x720_bgra.raw", (const uint8_t*)outFrame->data(), outFrame->size(), 0);
+}
+
+BOOST_AUTO_TEST_CASE(YUV444_scale_rotate)
+{
+	ImageMetadata::ImageType::YUV444;
+	AffineTestsStruct f("./data/yuv444.raw", 400, 400, ImageMetadata::ImageType::YUV444, size_t(0), AffineTransformProps::NN, 90, 0, 0, 2.5f);
+
+	f.fileReader->step();
+	f.copy1->step();
+	f.affineTransform->step();
+	f.copy2->step();
+
+	auto outputPinId = f.copy2->getAllOutputPinsByType(FrameMetadata::RAW_IMAGE_PLANAR)[0];
+	auto frames = f.m3->pop();
+	BOOST_TEST((frames.find(outputPinId) != frames.end()));
+	auto outFrame = frames[outputPinId];
+	BOOST_TEST(outFrame->getMetadata()->getFrameType() == FrameMetadata::RAW_IMAGE_PLANAR);
+	Test_Utils::saveOrCompare("./data/testOutput/affinetransform_tests_YUV444_scale_rotate_yuv444.raw", (const uint8_t*)outFrame->data(), outFrame->size(), 0);
+}
+BOOST_AUTO_TEST_CASE(YUV444_shift_scale_rotate)
+{
+	ImageMetadata::ImageType::YUV444;
+	AffineTestsStruct f("./data/yuv444_1920x1080.raw", 1920, 1080, ImageMetadata::ImageType::YUV444, size_t(0), AffineTransformProps::CUBIC2P_CATMULLROM,5, 200, 300, 2.0f);
+
+	f.fileReader->step();
+	f.copy1->step();
+	f.affineTransform->step();
+	f.copy2->step();
+
+	auto outputPinId = f.copy2->getAllOutputPinsByType(FrameMetadata::RAW_IMAGE_PLANAR)[0];
+	auto frames = f.m3->pop();
+	BOOST_TEST((frames.find(outputPinId) != frames.end()));
+	auto outFrame = frames[outputPinId];
+	BOOST_TEST(outFrame->getMetadata()->getFrameType() == FrameMetadata::RAW_IMAGE_PLANAR);
+	Test_Utils::saveOrCompare("./data/testOutput/affinetransform_tests_yuv444_1920x1080.raw", (const uint8_t*)outFrame->data(), outFrame->size(), 0);
+}
```

Comment:
add one test for dynamic change of properties aka getProps and setProps

Reply:
added